mava/wrappers/jumanji.py

```diff
@@ -311,6 +312,143 @@ def observation_spec(
         return specs.Spec(Observation, "ObservationSpec", **obs_data)
 
 
+def _slice_around(pos: chex.Array, fov: int) -> Tuple[chex.Array, chex.Array]:
+    """Return the start and length of a slice that when used to index a grid will
+    return a 2*fov+1 x 2*fov+1 sub-grid centered around pos.
+
+    Returns are meant to be used with a `jax.lax.dynamic_slice`
+    """
+    # Because we pad the grid by fov we need to shift the pos to the position
+    # it will be in the padded grid.
+    shifted_pos = pos + fov
+
+    start_x = shifted_pos[0] - fov
+    start_y = shifted_pos[1] - fov
+    return start_x, start_y
+
+
+# get location coordinates from 2D grid
+def _get_location(grid: chex.Array) -> chex.Array:
+    row_len = grid.shape[-1]
+    index = jnp.argmax(grid)
+    return jnp.asarray((jnp.floor(index / row_len), jnp.remainder(index, row_len)), dtype=int)
+
+
+class VectorConnectorWrapper(JumanjiMarlWrapper):
+    """Multi-agent wrapper for the MA Connector environment.
+    This env should have the AgentID wrapper applied to it since there is not longer a channel that
+    can encode AgentID information.
+    """
+
+    def __init__(self, env: MaConnector, add_global_state: bool = False):
+        super().__init__(env, add_global_state)
+        self._env: MaConnector
+        self.fov = 2
+
+    def modify_timestep(self, timestep: TimeStep) -> TimeStep[Observation]:
+        """Modify the timestep for the Connector environment."""
+
+        # TARGET = 3 = The number of different types of items on the grid.
+        def create_agents_view(grid: chex.Array) -> chex.Array:
+            positions = jnp.where(grid % TARGET == POSITION, True, False)
+            targets = jnp.where((grid % TARGET == 0) & (grid != EMPTY), True, False)
+            paths = jnp.where(grid % TARGET == PATH, True, False)
+
+            # group positions and paths
+            blockers = jnp.where(positions, 1, jnp.where(paths, -1, 0))
+
+            position_per_agent = jnp.where(grid == POSITION, True, False)
+            target_per_agent = jnp.where(grid == TARGET, True, False)
```

Comment:
No need for the `where`

```suggestion
            position_per_agent = grid == POSITION
            target_per_agent = grid == TARGET
```